### PR TITLE
Add support for ccr follow info api to HLRC.

### PR DESCRIPTION
**Pull request by martijnvg**
This API was recently introduced via #37408, but the hlrc side is still missing.
Relates to #33824

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ccr/FollowInfoResponse.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            @SuppressWarnings("unchecked")
+            List<FollowerInfo> infos = (List<FollowerInfo>) args[0];
+            return new FollowInfoResponse(infos);
+        });
+
+    static {
+        PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(), FollowerInfo.PARSER, FOLLOWER_INDICES_FIELD);
+    }
+
+
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ccr/FollowInfoResponse.java]: extra newline

File: client/rest-high-level/src/main/java/org/elasticsearch/client/CcrClient.java [hunk 2]
```diff
@@ -452,4 +454,47 @@ public void getFollowStatsAsync(FollowStatsRequest request,
...
+        restHighLevelClient.performRequestAsyncAndParseEntity(
+            request,
+            CcrRequestConverters::getFollowInfo,
+            options,
+            FollowInfoResponse::fromXContent,
+            listener,
+            Collections.emptySet()
+        );
+    }
+
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/CcrClient.java]: extra line

**Review by hub-cap — APPROVED** [review]
super easy to read. Youve done this before, i can tell ;)

Also, I just noticed (after all these reviews heh) that there is no `CCRRequestConvertersTests` class... Could you add that in a new PR or put up a GH ticket so someone can do it?

**Comment by martijnvg** [comment]
> Could you add that in a new PR or put up a GH ticket so someone can do it?

👍I will open a follow up pr for this.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Pull request merged**